content/en/blog/_posts/2020-05-14-my-exciting-journey-into-kubernetes-history.md

```diff
@@ -0,0 +1,656 @@
+> ./main export
+INFO | Getting GITHUB_TOKEN from environment variable
+INFO | Dumping all issues
+INFO | Pulling 90929 items
+INFO | 1: Unit test coverage in Kubelet is lousy. (~30%)
+INFO | 2: Better error messages if go isn't installed, or if gcloud is old.
+INFO | 3: Need real cluster integration tests
+INFO | 4: kubelet should know which containers it is managing
+… [just wait 8 hours] …
+```
+
+To update the data between the last time stamp stored in the repository we can
+run:
+
+```
+> ./main export --update-api
+INFO | Getting GITHUB_TOKEN from environment variable
+INFO | Retrieving issues and PRs
+INFO | Updating API
+INFO | Got update timestamp: 2020-05-09T10:57:40.854151
+INFO | 90786: Automated cherry pick of #90749: fix: azure disk dangling attach issue
+INFO | 90674: Switch core master base images from debian to distroless
+INFO | 90086: Handling error returned by request.Request.ParseForm()
+INFO | 90544: configurable weight on the CPU and memory
+INFO | 87746: Support compiling Kubelet w/o docker/docker
+INFO | Using already extracted data from data/data.pickle
+INFO | Loading pickle dataset
+INFO | Parsed 34380 issues and 55832 pull requests (90212 items)
+INFO | Updating data
+INFO | Updating issue 90786 (updated at 2020-05-09T10:59:43Z)
+INFO | Updating issue 90674 (updated at 2020-05-09T10:58:27Z)
+INFO | Updating issue 90086 (updated at 2020-05-09T10:58:26Z)
+INFO | Updating issue 90544 (updated at 2020-05-09T10:57:51Z)
+INFO | Updating issue 87746 (updated at 2020-05-09T11:01:51Z)
+INFO | Saving data
+```
+
+This gives us an idea how fast the project is actually moving: On a Saturday
+noon European time 5 issues and pull requests got updated within literally 5
+minutes!
+
+Funnily enough, [Joe Beda][32], one of the founders of Kubernetes, created the
+first GitHub issue [mentioning that the unit test coverage is too low][33]. The
+issue has no further description than the title and no enhanced labeling
+applied, like we know from more recent issues and pull requests. But now we have
+to explore the exported data more deeply to actually do something useful with
+it.
+
+[32]: https://github.com/jbeda
+[33]: https://github.com/kubernetes/kubernetes/issues/1
+
+# Exploring the Data
+
+Before we can start creating machine learning models and train them, we have to
+get an idea about how our data is structured and what we want to achieve in
+general.
+
+To get a better feeling about the amount of data, let’s look at how many issues
+and pull requests have been created over time inside the Kubernetes repository:
+
+```
+> ./main analyze --created
+INFO | Using already extracted data from data/data.pickle
+INFO | Loading pickle dataset
+INFO | Parsed 34380 issues and 55832 pull requests (90212 items)
+```
+
+The Python [matplotlib][40] module should pop up a graph which looks like this:
+
+![created-all](https://raw.githubusercontent.com/kubernetes-analysis/kubernetes-analysis/master/assets/created-all.svg)
```

Comment:
If you can, put the images in the website repo so the Netlify CDN can serve them.

Reply:
Also, pleas consider using spaces between words in alt attributes: this helps with accessibility.

Reply:
Changed it to spaces instead of dashes.

The files behind the images from the assets are updated on a regular basis, so I see this as kind of a feature that the blog post stays up-to-date over time. WDYT, should I still stick to the static images?

Reply:
My opinion is to use the Netlify CDN to host the static images, for now. That will prevent the blog post being out of sync with the images being shown (almost like pinning to a specific version vs. using `latest`).

Reply:
Sounds good to me, I added the images to a directory which is named exactly like the blog post. Let me know if you have a better location in mind within the repository :+1: